src/Knp/Menu/Matcher/Voter/RouteVoter.php

```diff
@@ -83,7 +83,7 @@ private function isMatchingRoute(array $testedRoute)
         $routeParameters = $this->request->attributes->get('_route_params', array());
 
         foreach ($testedRoute['parameters'] as $name => $value) {
-            if (!isset($routeParameters[$name]) || (string) $routeParameters[$name] !== (string) $value) {
+            if (!isset($routeParameters[$name]) || $routeParameters[$name] !== (string) $value) {
```

Comment:
@Nek- : what if `routeParameters[$name]` is an integer ? This question is not so hypothetical, it just happened to me ;)